packages/@aws-cdk/aws-glue/lib/table.ts

```diff
@@ -0,0 +1,239 @@
+import iam = require('@aws-cdk/aws-iam');
+import s3 = require('@aws-cdk/aws-s3');
+import cdk = require('@aws-cdk/cdk');
+import { IDatabase } from './database';
+import { CfnTable } from './glue.generated';
+import { Column } from './schema';
+import { StorageType } from './storage-type';
+
+export interface ITable extends cdk.IConstruct {
+  readonly tableArn: string;
+  readonly tableName: string;
+
+  export(): TableImportProps;
+}
+
+export interface TableImportProps {
+  tableArn: string;
+  tableName: string;
+}
+
+export interface TableProps {
+  /**
+   * Name of the table.
+   */
+  tableName: string;
+
+  /**
+   * Description of the table.
+   *
+   * @default generated
+   */
+  description?: string;
+
+  /**
+   * Database in which to store the table.
+   */
+  database: IDatabase;
+
+  /**
+   * S3 bucket in which to store data.
+   *
+   * @default one is created for you
+   */
+  bucket?: s3.Bucket;
+
+  /**
+   * S3 prefix under which table objects are stored.
+   *
+   * @default data/
+   */
+  prefix?: string;
+
+  /**
+   * Columns of the table.
+   */
+  columns: Column[];
+
+  /**
+   * Partition columns of the table.
+   *
+   * @default table is not partitioned
+   */
+  partitionKeys?: Column[]
+
+  /**
+   * Storage type of the table's data.
+   */
+  storageType: StorageType;
```

Comment:
Default to JSON?

Reply:
I think it's best to ask customers to be explicit about their data format. Front-load the important questions: schema, file format, location and security.